src/movies.js

```diff
@@ -1,16 +1,111 @@
+
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(arr) { 
+    return arr.map(function (item) {
+        return item.director;
+    })
+};
+
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(arr) {
+
+    return arr.filter(function(item) {
+      if (item.director === 'Steven Spielberg' && item.genre.includes('Drama')) {
+        return true;
+      }
+    }).length
+    
+  };
+  
```

Comment:
Great work Pedro! I just want to remind you that filter returns something as long as it is true. Therefore, you don't necessarily need an if statement. Your if statement here could also been written like this:
return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama').
If anything, i recommend you to check the MDN documentation.